lib/index.js

```diff
@@ -529,56 +443,31 @@ function _initDatabase(callback) {
           }, err));
       });
     },
-    openLocal: callback => {
-      if(!config.local.enable) {
-        logger.debug('initializing local database disabled');
-        return callback(null, null);
-      }
-      logger.info('initializing local database', {url: config.local.url});
-      _openDatabase({url: config.local.url, init: true}, (err, results) => {
-        db = results.db;
-        if(!err) {
-          return callback(null, true);
-        }
-        if(api.isDatabaseError(err) && (err.code === MDBE_AUTH_FAILED ||
-          err.message === 'could not authenticate')) {
-          // auth failed, either DB didn't exist or bad credentials
-          logger.info('local database authentication failed:' +
-            ' db=' + config.local.name +
-            ' username=' + config.username);
-          if(config.adminPrompt) {
-            return callback(null, false);
-          }
-        }
-        return callback(new BedrockError(
-          'Could not initialize local database.',
-          'DatabaseError', {
-            url: config.local.url
-          }, err));
-      });
-    },
-    check: ['open', 'openLocal', (results, callback) => {
+    check: ['open', (results, callback) => {
       // open and authenticated, finish
-      if(results.open && (results.openLocal || !config.local.enable)) {
+      if(results.open) {
         return callback();
       }
-      // try to create user
-      _createUser(client, db, callback);
+      // if the database did not open then the mongoDB server is running,
+      // but the user supplied incorrect information in the config.
+      const {config} = bedrock;
+      const {name, username, connectOptions = {}} = config.mongodb || {};
+      throw new Error('\nThe database "' + name +
+        '" failed to open. Please ensure the database ' + name +
+        ' exists, the primary user "' + username + '" ' +
+        'exists and has sufficient access privileges,' +
+        ' and that "connectOptions.authSource" ' +
+        connectOptions.authSource + ' is set correctly.', results);
```

Comment:
- Use BedrockError.
- Is `results` all that useful here?
- If constructing a string like this, maybe use backtick interpolated style: \`foo ${name} bar baz\`.
- However, if using a BedrockError could use a simpler English string and put param in a details object.

Reply:
marking this as resolved as this code is not in the project anymore. You are correct it should have been a `BedrockError` and I'm sorry for not using one.